src/ray/core_worker/transport/dependency_resolver.cc

```diff
@@ -119,7 +119,8 @@ void LocalDependencyResolver::ResolveDependencies(
       {
         absl::MutexLock lock(&mu_);
         state->local_dependencies[obj_id] = std::move(obj);
-        if (--state->obj_dependencies_remaining == 0) {
+        if (--state->obj_dependencies_remaining == 0 &&
```

Comment:
@stephanie-wang I needed to do this to fix `test_reference_counting.py::test_dependency_refcounts` (this is the same test that's been broken on windows). I'm wondering if this PR just made the bug more frequent on other platforms?

Reply:
Please revert this, should be fixed in #18862.